### ISSUE TITLE
[JavaScript] Incorrect TypeScript highlighting for inline declaration of function return value

### ISSUE THREAD

**Issue by IPWright83** [sublimehq/Packages#3441]
### What happened?

The following code highlights the `)` at the end of the return expression as being invalid when using the  TypeScript language syntax:

```
export const useIsOutsideRange = (
    startDateMoment?: Moment | null,
    focusedInput?: DayPickerRangeControllerShape['focusedInput'],
): ((day: Moment) => boolean) => {
    return useCallback(
        (day: Moment) => isOutsideRange(day, startDateMoment, focusedInput),
        [startDateMoment, focusedInput],
    );
};
```

![image](https://user-images.githubusercontent.com/1374775/177128567-7f8ffacb-4237-4386-a3b8-0d0f04aa9652.png)

It's fine if I remove the brackets like below, however we have a prettier rule that auto re-inserts them unfortunately:
```
export const useIsOutsideRange = (
    startDateMoment?: Moment | null,
    focusedInput?: DayPickerRangeControllerShape['focusedInput'],
): (day: Moment) => boolean => {
    return useCallback(
        (day: Moment) => isOutsideRange(day, startDateMoment, focusedInput),
        [startDateMoment, focusedInput],
    );
};
```

**Comment by alecmev** [comment]
Minimal repro:

```ts
(x): ((y) => any) => 42
```

**Comment by Thom1729** [comment]
Looks like I missed this when it came in. I'll take a look.

**Comment by Thom1729** [comment]
Should be fixed in #3455. Thanks both for the reports.
